lib/Doctrine/Common/Cache/PredisCache.php

```diff
@@ -45,8 +45,17 @@ protected function doFetch($id)
     protected function doFetchMultiple(array $keys)
     {
         $fetchedItems = call_user_func_array(array($this->client, 'mget'), $keys);
+        $foundItems = array();
+        $resultCounter = 0;
+
+        foreach ($keys AS $key) {
+            if ($fetchedItems[$resultCounter] !== null) {
+                $foundItems[$key] = unserialize($fetchedItems[$resultCounter]);
+            }
+            $resultCounter++;
+        }
 
-        return array_filter(array_combine($keys, array_map('unserialize', $fetchedItems)));
```

Comment:
You could actually `array_filter()` before passing to `unserialize()` instead of re-writing the whole body as it was above